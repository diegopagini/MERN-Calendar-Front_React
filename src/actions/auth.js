/** @format */
import { fetchSinToken, fetchConToken } from '../helpers/fetch';
import { types } from '../types/types';
import Swal from 'sweetalert2';

export const startLogin = (email, password) => {
	return async (dispatch) => {
		const resp = await fetchSinToken('auth', { email, password }, 'POST');
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			Swal.fire('Error', body.msg, 'error');
		}
	};
};

export const startRegister = (email, password, name) => {
	return async (dispatch) => {
		const resp = await fetchSinToken(
			'auth/new',
			{ email, password, name },
			'POST'
		);
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			Swal.fire('Error', body.msg, 'error');
		}
	};
};

export const startChecking = () => {
	return async (dispatch) => {
		const resp = await fetchConToken('auth/renew');
		const body = await resp.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem('token-init-date', new Date().getTime());

			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			dispatch(checkingFinish());
		}
	};
};

const login = (user) => ({
	type: types.authLogin,
	payload: user,
});

const checkingFinish = () => ({
	type: types.authCheckingFinish,
});

export const startLogout = () => {
	return (dispatch) => {
		localStorage.clear();
		dispatch(eventLogout());
		dispatch(logout());
	};
};

const logout = () => ({
	type: types.authLogout,
});

export const eventLogout = () => ({
	type: types.eventLogout,
});
