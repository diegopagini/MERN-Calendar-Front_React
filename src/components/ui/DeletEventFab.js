/** @format */

import React from 'react';
import { useDispatch } from 'react-redux';
import { eventStartDelete } from '../../actions/events';

export const DeletEventFab = () => {
	const dispatch = useDispatch();

	const handleDelete = () => {
		dispatch(eventStartDelete());
	};

	return (
		<button
			className='btn btn-danger fab-danger animate__animated animate__fadeInLeft'
			onClick={handleDelete}
		>
			<i className='fas fa-trash'></i>
			<span> Borrar Evento</span>
		</button>
	);
};
