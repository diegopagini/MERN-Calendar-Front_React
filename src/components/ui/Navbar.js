/** @format */
import React from 'react';
import { Offline, Online } from 'react-detect-offline';
import { useDispatch, useSelector } from 'react-redux';

import { startLogout } from '../../actions/auth';

export const Navbar = () => {
	const { name } = useSelector((state) => state.auth);
	const dispatch = useDispatch();

	const handleLogout = () => {
		dispatch(startLogout());
	};

	return (
		<div className='navbar navbar-dark bg-dark mb-4'>
			<span className='navbar-brand'>{name}</span>

			<Online>
				<span className='text-success'>Online</span>
			</Online>
			<Offline>
				<span className='text-danger'>Offline</span>
			</Offline>

			<button onClick={handleLogout} className='btn btn-outline-danger'>
				<i className='fas fa-sign-out-alt'></i>
				<span> Salir</span>
			</button>
		</div>
	);
};
