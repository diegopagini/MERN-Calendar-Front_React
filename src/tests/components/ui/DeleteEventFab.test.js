/** @format */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

jest.mock('../../../actions/events', () => ({
	eventStartDelete: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<DeleteEventFab />
	</Provider>
);

describe('Pruebas en <DeleteEventFab />', () => {
	test('debe de mostrarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de llamar el eventStartDelete al hacer click', () => {
		wrapper.find('button').prop('onClick')();

		expect(eventStartDelete).toHaveBeenCalled();
	});
});
