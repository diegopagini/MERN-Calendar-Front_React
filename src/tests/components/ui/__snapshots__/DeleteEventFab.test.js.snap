// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DeleteEventFab /> debe de mostrarse correctamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar evento 
  </span>
</button>
`;
